### PR TITLE
Incorporate openwrt patches

### PR DESCRIPTION
**Pull request by therealbstern**
This addresses #50 and applies the relevant openwrt patches.  Some were moot, some had been applied, some looked irrelevant.

I also fixed a build warning in `input_raspicam.c` which got committed to this branch by accident but seems harmless.

### CODE REVIEW

**Comment by roger-** [comment]
Thanks for this!

Why were some of the other patches irrelevant BTW?

**Comment by therealbstern** [comment]
They were irrelevant because they were already included in this fork.  Possibly OpenWRT got them from here, or both sets of developers saw the same problem(s).

**Comment by therealbstern** [comment]
Ignore my previous - not enough coffee.  The irrelevant ones were because they apply to files that do not appear in this fork, if I recall correctly.

**Comment by roger-** [comment]
Ah ok. I'll have to identify and combine them together on the OpenWrt end.

**Comment by therealbstern** [comment]
Oh!  I hope I didn't make more work for you.  I named my commits after the relevant OpenWRT patches, so maybe that will help identify the ones I used.

**Comment by roger-** [comment]
No worries, looks like the ones you skipped indeed aren't relevant since the switch to cmake.  Thanks again!

**Comment by roger-** [comment]
@jacksonliam could you take a look at this?

File: mjpg-streamer-experimental/plugins/input_uvc/v4l2uvc.c [hunk 1]
```diff
@@ -383,11 +386,16 @@ static int init_v4l2(struct vdIn *vd)
...
-                        // set FPS to maximum in order to minimize the lagging
                         memset(setfps, 0, sizeof(struct v4l2_streamparm));
                         setfps->type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
                         setfps->parm.capture.timeperframe.numerator = 1;
-                        setfps->parm.capture.timeperframe.denominator = 255;
+                        if (vd->formatIn == V4L2_PIX_FMT_MJPEG) {
+                            // set FPS to maximum in order to minimize the lagging
+                            setfps->parm.capture.timeperframe.denominator = 255;
+                        } else {
+                            setfps->parm.capture.timeperframe.denominator = vd->fps;
```
Comment (jacksonliam) [inline mjpg-streamer-experimental/plugins/input_uvc/v4l2uvc.c]: Doing this differently depending on the hardware's output format does not seem right to me. Any ideas on why we'd want to do this?

I'm not sure if asking for vd->fps is the right thing here, its a recipie for a jerky picture / aliasing. I'm not sure if the original solution of setting it to 255 is right either. 

I don't like this whole 'automagic software framedrop thing' anyway and am concidering removing it.

**Comment by roger-** [comment]
@jacksonliam perhaps the author of these patches (@thess) can address your concerns.

**Comment by thess** [comment]
Sorry, I cannot take credit for these patches. You might find some more info about them here:
https://github.com/openwrt/packages/commit/b17a973f141cb6f5c1d2f79b3aac9a68aac02679

**Comment by roger-** [comment]
My mistake, thanks @thess!

File: mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c [hunk 2]
```diff
@@ -659,8 +664,8 @@ void *cam_thread(void *arg)
         if (pcontext->videoIn->soft_framedrop == 1) {
             unsigned long last = pglobal->in[pcontext->id].timestamp.tv_sec * 1000 +
                                 (pglobal->in[pcontext->id].timestamp.tv_usec/1000); // convert to ms
-            unsigned long current = pcontext->videoIn->buf.timestamp.tv_sec * 1000 +
-                                    pcontext->videoIn->buf.timestamp.tv_usec/1000; // convert to ms
+            unsigned long current = pcontext->videoIn->tmptimestamp.tv_sec * 1000 +
+                                    pcontext->videoIn->tmptimestamp.tv_usec/1000; // convert to ms
```
Comment (roger-) [inline mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c]: Please see [here](https://github.com/oliv3r/mjpg-streamer/commit/11b28b36a8711b53658e8bbc50435595522f91ba) for some details from the author of this patch.

File: mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c [hunk 3]
```diff
@@ -378,7 +378,12 @@ int input_init(input_parameter *param, int id)
         IPRINT("not enough memory for videoIn\n");
         exit(EXIT_FAILURE);
     }
-    
+
+    /* Non-MJPEG formats seem to fail with unlimited FPS */
+    if ((format != V4L2_PIX_FMT_MJPEG) && (fps == -1)) {
+        fps = 15;
```
[the remark this answers is not in the document]
Comment (therealbstern) [reply mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c]: Somehow this completely fell off the table.  I apologize.

Anyway, this came from the openwrt patches, I have no idea why they thought that.  I will remove it.

File: mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (therealbstern) [reply mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c]: Sadly, this has evaporated and it's not in the wayback machine.

File: mjpg-streamer-experimental/plugins/input_uvc/v4l2uvc.c (hunk 1 above)
Comment (therealbstern) [reply mjpg-streamer-experimental/plugins/input_uvc/v4l2uvc.c]: Will revert.

File: mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (therealbstern) [reply mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c]: Looks like this is already in your tree, so I'm going to assume it's fine.

**Comment by therealbstern** [comment]
This has gotten way out of sync with master.  I'm going to try to rebase against a more current pull.

**Pull request closed without merging**